src/main.rs

```diff
@@ -102,16 +115,28 @@ async fn provision() -> Result<(), anyhow::Error> {
     )?;
     default_headers.insert(header::USER_AGENT, user_agent);
     let client = Client::builder()
-        .timeout(std::time::Duration::from_secs(30))
+        .timeout(std::time::Duration::from_secs(HTTP_TIMEOUT_SEC))
         .default_headers(default_headers)
         .build()?;
 
+    let imds_endpoint = azure_core::Url::parse(IMDS_ENDPOINT)?;
+    let imds_client = ImdsClient::builder(Arc::new(AzureCliCredential::new()))
```

Comment:
does this require authentication to work? provisioning shouldn't require obtaining credential